MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesDataFactory.java

```diff
@@ -0,0 +1,29 @@
+package com.novoda.demo.movies;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.paging.DataSource;
+
+import com.novoda.demo.movies.model.Movie;
+
+public class MoviesDataFactory extends DataSource.Factory<Integer, Movie> {
+
+    private final MovieService movieService;
+    private final MutableLiveData<MoviesDataSource> dataSource;
```

Comment:
The `MoviesDataSource` owns the information about the `NetworkStatus, but since the framework decides when to create it, we need to observe it.